flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/gsa/GatherSumApplyIteration.java

```diff
@@ -289,6 +300,11 @@ private GatherUdf(GatherFunction<VV, EV, M> gatherFunction, TypeInformation<Tupl
 
 		@Override
 		public void open(Configuration parameters) throws Exception {
+			try {
+				Collection<LongValue> numberOfVertices = getRuntimeContext().getBroadcastVariable("number of vertices");
+				this.gatherFunction.setNumberOfVertices(numberOfVertices.iterator().next().getValue());
+			} catch (Exception e) {
+			}
```

Comment:
The BatchAPI usually throws exceptions when people build a seemingly incorrect program, for example by using a broadcast variable that does not exist. It seemed to help people to catch bugs faster.

The exception makes it obvious that something is wrong, the empty set would make various programs run but return a wrong result. That is much harder to recognize and debug.